src/Lib.P2P/Cid.cs

```diff
@@ -0,0 +1,610 @@
+    [JsonConverter(typeof(CidJsonConverter))]
+    public class Cid : IEquatable<Cid>
+    {
+        /// <summary>
+        ///   The default <see cref="ContentType"/>.
+        /// </summary>
+        public const string DefaultContentType = "dag-pb";
+
+        private string encodedValue;
+        private int version;
+        private string encoding = MultiBase.DefaultAlgorithmName;
+        private string contentType = DefaultContentType;
+        private MultiHash hash;
+
+        /// <summary>
+        ///   Throws if a property cannot be set.
+        /// </summary>
+        /// <exception cref="NotSupportedException">
+        ///   When a property cannot be set.
+        /// </exception>
+        /// <remarks>
+        ///   Once <see cref="Encode"/> is invoked, the CID's properties
+        ///   cannot be set.
+        /// </remarks>
+        private void EnsureMutable()
+        {
+            if (encodedValue != null) throw new NotSupportedException("CID cannot be changed.");
+        }
+
+        /// <summary>
+        ///   The version of the CID.
+        /// </summary>
+        /// <value>
+        ///   0 or 1.
+        /// </value>
+        /// <remarks>
+        ///   <para>
+        ///   When the <see cref="Version"/> is 0 and the following properties
+        ///   are not matched, then the version is upgraded to version 1 when any 
+        ///   of the properties is set.
+        ///   <list type="bullet">
+        ///   <item><description><see cref="ContentType"/> equals "dag-pb"</description></item>
+        ///   <item><description><see cref="Encoding"/> equals "base58btc"</description></item>
+        ///   <item><description><see cref="Hash"/> algorithm name equals "sha2-256"</description></item>
+        ///   </list>
+        ///   </para>
+        ///   <para>
+        ///   </para>
+        ///   The default <see cref="Encoding"/> is "base32" when the
+        ///   <see cref="Version"/> is not zero.
+        /// </remarks>
+        public int Version
+        {
+            get => version;
+            set
+            {
+                EnsureMutable();
+                if (version == 0 && value > 0 && Encoding == "base58btc") encoding = "base32";
+                version = value;
+            }
+        }
+
+        /// <summary>
+        ///   The <see cref="MultiBase"/> encoding of the CID.
+        /// </summary>
+        /// <value>
+        ///   base58btc, base32, base64, etc.  Defaults to <see cref="MultiBase.DefaultAlgorithmName"/>.
+        /// </value>
+        /// <remarks>
+        ///    Sets <see cref="Version"/> to 1, when setting a value that
+        ///    is not equal to "base58btc".
+        /// </remarks>
+        /// <seealso cref="MultiBase"/>
+        public string Encoding
+        {
+            get => encoding;
+            set
+            {
+                EnsureMutable();
+                if (Version == 0 && value != "base58btc") Version = 1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)